### PR TITLE
Add support to pre-allocate the meta area

### PR DESCRIPTION
**Pull request by urso**
- Move file options to separate opts.go file
- Add validation on file options
- Add support to pre-allocate the meta area at file creation time
- Add test for file with initial meta area

### CODE REVIEW

File: opts.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	// PageSize sets the files page size on file creation. PageSize is ignored if
+	// the file already exists.
+	// If PageSize is not configured, the OSes main memory page size is selected.
+	PageSize uint32
+
+	// InitMetaArea configures the minimum amount of page in the meta area.
+	// The amount of pages is only allocated when the file is generated.
+	// The meta area grows by double the current meta area. To reduce the
+	// total amount of pages moved to meta area on grow, it is recommended that
+	// the value of InitMetaArea is a power of 2.
```
Comment (ph) [inline opts.go]: To be honest I think we should enforce that? WDYT?

File: opts.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		totalPages := o.MaxSize / uint64(o.PageSize)
+		avail := totalPages - headerPages
+		if uint64(metaSz) >= avail {
+			return fmt.Errorf("meta area of %v pages exceeds the available pages %v",
+				metaSz, avail)
+		}
+	}
+
+	return nil
+}
```
Comment (ph) [inline opts.go]: Ty for the validation!

File: opts.go (hunk 1 above)
Comment (urso) [reply opts.go]: I didn't want to enforce it. E.g. assuming we don't have much fragmentation, the `pq` will run fine with 10 pages being preallocated. Even when growing, less pages might be used in the end, then starting with 16. Instead I will add a grow factor that must be > 1 in the future.

Comment (ph) [reply opts.go]: Sound good!

**Pull request merged**